### DISCUSSION TITLE
Supabase as a backend for a CMS

### DISCUSSION THREAD

**Discussion by linusdm** [supabase/supabase#7540] [Questions]
Hi Supabase community 👋

I was wondering if anyone is dabbling with using supabase as a backend for an extended, multi-site CMS. Many of the supabase features seem very appropriate for this kind of challenge (rest api, RLS, easy data setup and migrations, edge functions for more complex integrations, change callbacks for hooking up business rules, storage for other kinds of assets).

The only part missing is a rendering pipeline from customer facing templates + data (from the database) to html. And maybe some management features in a nice backend UI. I don't know exactly how this would work, but it seems doable.

Does this sound sane? Are there already some people doing this kind of thing with supabase?

Googling firbase + CMS does spawn a few projects, so at least there is some evidence of the sanity of this idea 🙃

Thanks in advance for any thoughts in this topic! 🙏

**Comment by burggraf** [comment]
Can you give some more details about what you mean by CMS?  That's a pretty broad term that can be used for a wide range of project types.  People are using Supabase as a backend for all sorts of projects that have CMS features.  What are your specific goals?

**Reply by linusdm** [reply]
Hey. At its core it would mean a way to transform a set of structured data into nice looking pages, with the help of templates. A back-end would provide means to edit the data, and the templates. But in the other direction a CMS would also capture (form) data and feed it into the back-end data store (and validate it).

**Reply by joaoflaviosantos** [reply]
See this: https://github.com/jordienr/SupaCMS

**Comment by bastianhilton** [comment]
I'm working on one now https://github.com/meeovi/alternatecms.

**Comment by Charuru** [comment]
How hard would it be to use it with an existing CMS like directus?

**Reply by Charuru** [reply]
NVM found this, looks like it's an officially suggested pairing: https://supabase.com/docs/guides/integrations/directus

**Reply by Mlamalerie** [reply]
Did you manage to configure all this and synchornise supabse and directus? 😭

**Comment by softmarshmallow** [comment]
Woking on this as my full-time job !

![](https://github.com/gridaco/grida/raw/main/.readme/cover.png)

https://github.com/gridaco/grida


- No SQL or Coding required (Not even connects via db url)
- Supports Supabae Storage (And manage them like notion)
- Write blog contents, get form responses (without any extra layer, just your supabase project)

Still in beta, so feel free to reach me for any questions or onboarding help ; )


(didn't mean to spam with the image, its what I have for now 😥)

**Comment by amrutha97** [comment]
Try https://github.com/StructuredLabs/supawald
